packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.styles.ts

```diff
@@ -252,28 +252,24 @@ export const getStyles = (props: IDetailsRowStyleProps): IDetailsRowStyles => {
           borderRight: `${cellStyleProps.cellRightPadding}px solid ${colors.defaultBackgroundColor}`,
           borderBottom: `${(values.rowHeight - values.rowShimmerIconPlaceholderHeight) / 2}px solid ${colors.defaultBackgroundColor}`,
           borderTop: `${(values.rowHeight - values.rowShimmerIconPlaceholderHeight) / 2}px solid ${colors.defaultBackgroundColor}`
-        }
-      }
-    },
+        },
 
-    isSelected && {
-      selectors: {
-        '&.$isRowHeader': {
+        '.is-selected > &.$isRowHeader': {
```

Comment:
@Vitalius1 this is where I need your expertise.

Currently, we are creating a new style for when `isSelected` is `true`. However, the parent component has `.is-selected` class applied which I'd like to use instead. This way the `.ms-DetailsRow-cell` class doesn't churn causing the component to unnecessarily re-render.

Reply:
Not sure these existing styles work currently... here's the mismatch:

![image](https://user-images.githubusercontent.com/706967/56525139-6d051b00-64ff-11e9-89a1-bae446349ee1.png)

The output selectors have `$` which does not seem correct. cc: @Vitalius1

Reply:
@Vitalius1 and I spoke yesterday, many of these `isRowHeader` selectors do not work as-is, particularly the one currently failing screenshots above.

I have refactored `is-selected` and high-contrast mode styles in commits https://github.com/OfficeDev/office-ui-fabric-react/pull/8799/commits/fff6dce78048185b62dd2e450208c1ca49d148b4 and https://github.com/OfficeDev/office-ui-fabric-react/pull/8799/commits/a74b32b3ab0d499c3882a5f5fbf2150c5c653f25

The selectors do not trigger React re-render (the goal of this PR) and fix the following high-contrast styling issue:

**Before:**

Only the background spanning the column cells is highlighted when selected.

![image](https://user-images.githubusercontent.com/706967/56608436-9a190280-65bf-11e9-9bdd-985da4f11cca.png)

**After:**

The entire row background is highlighted when selected.

![image](https://user-images.githubusercontent.com/706967/56608392-840b4200-65bf-11e9-8d52-0eac7c633616.png)